objects/src/notes/origin.rs

```diff
@@ -0,0 +1,35 @@
+use super::{BlockHeader, NoteError, NOTE_TREE_DEPTH};
+use crypto::merkle::NodeIndex;
+
+/// Represents the origin of a note.  This includes the block number the note was created in, and
+/// the node index of the note in the note Merkle tree of that block.  The note origin is used by
+/// the transaction kernel to verify the authenticity of a note.
+#[derive(Debug, Eq, PartialEq)]
+pub struct NoteOrigin {
+    block: BlockHeader,
+    node_index: NodeIndex,
+}
```

Comment:
Technically, we don't need the whole header here, right? I think, we only need the `note_root` and `core_hash`.

Also, we should probably store the full Merkle path here from the note leaf to the `note_root`.

Reply:
That's correct, we need `block_num`, `note_root` and `sub_hash`.

Reply:
I've updated this to only include required data.